src/components/abstractComponent.ts

```diff
@@ -138,18 +138,14 @@ export module Component {
           throw new Error("null arguments cannot be passed to _computeLayout() on a non-root node");
         }
       }
-      this._xOrigin = xOrigin;
-      this._yOrigin = yOrigin;
       var requestedSpace = this._requestedSpace(availableWidth, availableHeight);
       this._width  = this._isFixedWidth()  ? Math.min(availableWidth , requestedSpace.width)  : availableWidth ;
       this._height = this._isFixedHeight() ? Math.min(availableHeight, requestedSpace.height) : availableHeight;
 
-      var xPosition = this._xOrigin + this._xOffset;
-      var yPosition = this._yOrigin + this._yOffset;
-      xPosition += (availableWidth - this.width()) * this._xAlignProportion;
-      yPosition += (availableHeight - requestedSpace.height) * this._yAlignProportion;
+      this._xOrigin = xOrigin + this._xOffset + (availableWidth - this.width()) * this._xAlignProportion;
+      this._yOrigin = yOrigin + this._yOffset + (availableHeight - requestedSpace.height) * this._yAlignProportion;;
```

Comment:
Might want to refactor the parameters xOrigin/yOrigin.  In this context, now we have xOrigin be some computation of xOrigin amongst other parameters, which is semantically confusing.